app/api/route/message_api.rb

```diff
@@ -0,0 +1,31 @@
+module Route
+  class MessageApi < Base
+    paginate paginate_settings
+
+    params do
+      optional :thread, type: Integer, desc: 'ID of thread'
+      optional :order_by, type: Symbol, values:%i( created_at  ), desc: 'Order of returned issues.'
+      optional :order, type: Symbol, values:%i(asc desc), default: :asc
+      optional :submit_external, type: String, desc: 'Filter by external service submission mark'
+    end
+
+    helpers do
+      def post_or_get_messages
+        scope = Message.joins(:thread).includes(:created_by)
+        scope = scope.where('message_threads.privacy' => 'public')
```

Comment:
I'd prefer
```ruby
scope = scope.merge(MessageThread.is_public)
```
also we need `.approved` or we might show spam messages

Reply:
OK